swarm/api/api.go

```diff
@@ -93,6 +117,7 @@ func (self *Api) Resolve(uri *URI) (storage.Key, error) {
 	// if DNS is not configured, check if the address is a hash
 	if self.dns == nil {
 		if !isHash {
+			apiResolveFail.Inc(1)
```

Comment:
Again is this really a resolve failure? I'd expect the resolve failure metric to tell me how many times Swarm tried to actually resolve a name but failed. Perhaps my expectation is incorrect though?

Reply:
*My* expectation was to capture how many times a resolve was attempted with the meaning of requested (count) and how many times these attempts failed, answering the question: "How many times did people request a resolution, and how many times this request failed?". Both expectations (mine and yours) can lead to mal-interpretations. We should just settle on one. I believe it makes sense to keep them for now, and then remove such metrics as we go when we realize we are not gaining anything from them. Still, if people think we should remove ones like this one right away, I am ok with it.

Reply:
> I believe it makes sense to keep them for now, and then remove such metrics as we go when we realize we are not gaining anything from them

I think it should be the other way around. Just add metrics we know are useful, with the freedom to add more useful ones in the future.

In this specific case, if I wanted to know how many resolve failures I'm getting, I'd also want to know what domains are failing too, in which case I'd probably just look for errors in my logs (which have much more context than just a single number).

Reply:
When looking at failures, specifically in a production environment, but also in a staging/test environment, I would structure my workflow in creating alarms and visualizing error situations in a general manner first, like creating alarms when reaching certain thresholds. Only *then* I would, if the counters/graphs show unacceptable levels or alarms get triggered, I would then start the debugging process of looking at logs for why the alarms were triggered or why there is an unusual/unexpected high amount of errors. I believe this to be a very valid argument to keep this counter. Nevertheless, if it's still maintained that the counter itself is of no value, I will proceed to remove it.